tests/test_regrid/test_structured.py

```diff
@@ -0,0 +1,370 @@
+import numpy as np
+import pytest
+import xarray as xr
+
+from xugrid.regrid.structured import StructuredGrid1d, StructuredGrid2d
+
+# Testgrids
+# --------
+# grid a(x):               |______50_____|_____100_____|_____150_____|               -> source
+# grid b(x):        |______25_____|______75_____|_____125_____|_____175_____|        -> target
+# --------
+# grid aa(x):                       |______80_____|_____130_____|_____180_____|      -> source
+# grid bb(x):            |______40_____|______90_____|_____140_____|____190_____|    -> target
+# --------
+# grid a(x):               |______50_____|_____100_____|_____150_____|               -> source
+# grid bbb(x):              |__30__|__55__|__80_|__105__|                            -> target
+# --------
+
+
+@pytest.fixture
+def grid_a():
+    return xr.DataArray(
+        data=np.arange(9).reshape((3, 3)),
+        dims=["y", "x"],
+        coords={
+            "y": np.array([150, 100, 50]),
+            "x": np.array([50, 100, 150]),
+            "dx": 50.0,
+            "dy": -50.0,
+        },
+    )
+
+
+@pytest.fixture
+def grid_aa():
+    return xr.DataArray(
+        data=np.arange(9).reshape((3, 3)),
+        dims=["y", "x"],
+        coords={
+            "y": np.array([150, 100, 50]),
+            "x": np.array([80, 130, 180]),
+            "dx": 50.0,
+            "dy": -50.0,
+        },
+    )
+
+
+@pytest.fixture
+def grid_b():
+    return xr.DataArray(
+        data=np.arange(16).reshape((4, 4)),
+        dims=["y", "x"],
+        coords={
+            "y": np.array([175, 125, 75, 25]),
+            "x": np.array([25, 75, 125, 175]),
+            "dx": 50.0,
+            "dy": -50.0,
+        },
+    )
+
+
+@pytest.fixture
+def grid_bb():
+    return xr.DataArray(
+        data=np.arange(16).reshape((4, 4)),
+        dims=["y", "x"],
+        coords={
+            "y": np.array([175, 125, 75, 25]),
+            "x": np.array([40, 90, 140, 190]),
+            "dx": 50.0,
+            "dy": -50.0,
+        },
+    )
+
+
+@pytest.fixture
+def grid_bbb():
+    return xr.DataArray(
+        data=np.arange(16).reshape((4, 4)),
+        dims=["y", "x"],
+        coords={
+            "y": np.array([175, 125, 75, 25]),
+            "x": np.array([30, 55, 80, 105]),
+            "dx": 25.0,
+            "dy": -50.0,
+        },
+    )
+
+
+@pytest.fixture
+def grid_a_1d(grid_a):
```

Comment:
please also make a test case with a flipped 1d grid- for example StucturedGrid1d(grid_a, "y"). Then we can test operations on 2 1d grids, where 1 is flipped and the other is not.